superdesk/media/video.py

```diff
@@ -42,3 +44,161 @@ def get_meta(filestream):
+        "Title": photo.get("ObjectName", photo.get("Title")),
+        "Creator": photo.get("By-line", photo.get("Creator")),
+        "AuthorsPosition": photo.get("By-lineTitle", photo.get("CreatorsJobtitle")),
+        "Rights": photo.get("CopyrightNotice"),
+        "City": photo.get("City"),
+        "Country": photo.get("Country-PrimaryLocationName", photo.get("Country")),
+        "CountryCode": photo.get("Country-PrimaryLocationCode", photo.get("CountryCode")),
+        "Credit": photo.get("Credit", photo.get("CreditLine")),
+        "State": photo.get("Province-State", photo.get("ProvinceState")),
+        "Location": photo.get("Sub-location"),
+        "CreatorContactInfo": photo.get("Contact"),
+        "Language": photo.get("LanguageIdentifier"),
+        "Destination": photo.get("Destination"),
+        "ServiceIdentifier": photo.get("ServiceIdentifier"),
+        "ProductID": photo.get("ProductID"),
+        "DateSent": photo.get("DateSent"),
+        "TimeSent": photo.get("TimeSent"),
+        "EditStatus": photo.get("EditStatus"),
+        "Urgency": photo.get("Urgency"),
+        "SubjectCode": photo.get("SubjectReference"),
+        "Category": photo.get("Category"),
+        "SupplementalCategories": photo.get("SupplementalCategories"),
+        "Subject": photo.get("Keywords"),
+        "LocationCode": photo.get("ContentLocationCode"),
+        "LocationName": photo.get("ContentLocationName"),
+        "ReleaseDate": photo.get("ReleaseDate"),
+        "ReleaseTime": photo.get("ReleaseTime"),
+        "ExpirationDate": photo.get("ExpirationDate"),
+        "ExpirationTime": photo.get("ExpirationTime"),
+        "TimeCreated": photo.get("TimeCreated"),
+        "Source": photo.get("Source"),
+        **(
+            {"DateCreated": f"{photo.get('DateCreated')}T{photo.get('TimeCreated')}"}
+            if photo.get("DateCreated") and photo.get("TimeCreated")
+            else {}
+        ),
+    }
+    tags = {k: vv for k, v in xmp.items() if (vv := v or photo.get(k))}
+    return cast(VideoMetadata, tags)
+
+
+def map_exiftool_args(video: VideoMetadata) -> list[str]:
+    args = ["-sep", ","]
+    args.extend(f"-{k}={','.join(v) if isinstance(v, list) else v}" for k, v in video.items())
+    args.append("-overwrite_original")
+    return args
+
+
+def write_with_exiftool(bin: bytes, args: list[str]) -> bytes:
+    import tempfile
+    from exiftool import ExifToolHelper
+    from exiftool.exceptions import ExifToolExecuteError
+    from os import remove
+
+    with tempfile.NamedTemporaryFile(delete=False) as tmp:
+        tmp.write(bin)
+        tmp.flush()
+        path = tmp.name
+
+    try:
+        with ExifToolHelper() as et:
+            et.execute(*args, path)
+
+        with open(path, "rb") as r:
+            return r.read()
+    except ExifToolExecuteError as e:
+        logger.exception(e.stderr)
+        return bin
+    finally:
+        remove(path)
```

Comment:
I guess you could do the whole thing inside that tempfile block,
calling `tmp.seek(0)` after `et.execute` and returning `tmp.read()` after